### ISSUE TITLE
Can't run tests on 4.02.x

### ISSUE THREAD

**Issue by ostera** [ocaml/odoc#205]
After #198 we are required to pin the development version of `tyxml`, as [indicated here](https://github.com/ocaml/odoc/pull/198#issue-217566933).

However, when using the 4.02 switch, I'm seeing the following warnings from `opam`, that look harmless but end up in a very different generated HTML.

```
ostera/odoc λ opam switch 4.02.3+buckle-master
ostera/odoc λ opam pin add --dev-repo tyxml
[tyxml.dev] no changes from git+https://github.com/ocsigen/tyxml.git
tyxml is now pinned to git+https://github.com/ocsigen/tyxml.git (version dev)

The following dependencies couldn't be met:
  - tyxml → ocaml >= 4.03 → ocaml-variants >= 4.03.0
      base of this switch (use `--unlock-base' to force)
  - tyxml → ocaml >= 4.03 → ocaml-system >= 4.03.0
      unmet availability conditions, e.g. sys-ocaml-version = "4.03.0"
  - tyxml → ocaml >= 4.03 → ocaml-base-compiler >= 4.03.0
      conflict with the base packages of this switch

[NOTE] Pinning command successful, but your installed packages may be out of sync.
```

@aantron maybe this is something to report in the `tyxml` repo, but I've also read somewhere (I think among these issues) that we want to drop this dependency?

AFAICT, the current status is that the master branch can't be tested under 4.02 anymore.

**Comment by ostera** [comment]
It's also completely possible that I'm doing something wrong 😅

**Comment by ostera** [comment]
Okay, now after re-pinning it's working.

Nevermind!

**Comment by aantron** [comment]
@oxtera, TyXML `master` requires OCaml 4.03: https://github.com/ocsigen/tyxml/blob/3b7ddbf837349b17f310f241631d68c1efc98e46/tyxml.opam#L23

But I've opened a PR on TyXML to port it back to 4.02.3, until BuckleScript is ported to a newer version: https://github.com/ocsigen/tyxml/pull/214

So, you should pin either that branch of TyXML, or use the latest released version, which still supports 4.02.
